Update this sample app to work on cloud.gov

fureigh opened the pull request:
**User stories:**

1. As a potential customer evaluating cloud.gov, I want to be able to evaluate evidence that I'll be able to deploy my Drupal app(s) to cloud.gov. (Ideally, I want that evidence to be reassuring.)
2. As someone trying out Drupal on cloud.gov, I want any Drupal-related documentation to be accurate so I can deploy my Drupal app with minimal stress and hassle. (Applies to sandbox and paying users alike.)
3. As a cloud.gov team member, I want up-to-date documentation for deploying Drupal so that Drupal-using customers will be happier: less likely to need other support channels and more likely to be confident that cloud.gov supports them and meets their needs (i.e., that they're not an edge case).

**Technical notes:**
Pieces to contribute upstream:

1. https://github.com/18F/cf-ex-drupal/pull/1/commits/c681d5165a3d016c11e43828e4c4cdae50ecb7fc
2. 972c8eb4386b7183f92a6ad05791134a098856f0 (written more flexibly with conditionally specified index names, and acknowledging https://github.com/cloudfoundry-samples/cf-ex-drupal/issues/3)

fureigh committed:
Update part of README to be cloud.gov-specific

Prospective cloud.gov users trying out the platform aren't
necessarily already familiar with Cloud Foundry. By updating the
README to be cloud.gov-specific, we can reassure them that they're
in the right place and increase the instructions' applicability to
and accuracy on cloud.gov.

 * Change git repo address
 * Change some instances of `Cloud Foundry` to `cloud.gov`
 * Standardize to `buildpack`
 * Add URLs to PHP Buildpack references
 * Add `htdocs` per 6bcd8b37ebd0b79cea54945eb55ee69bac022861

fureigh committed:
Make numbered Markdown lists contiguous

fureigh commented:
This now has a corresponding [Favro card](https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6009).

fureigh committed:
Specify that this is Drupal 7

At the time of this writing, Drupal 8 is on the rise and Drupal 6
is still hanging on. To reduce unnecessary room for confusion,
let's be explicit that this example uses Drupal 7.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6009

fureigh committed:
Update db service name in manifest.yml

Previously this didn't match the example provided in the README,
so `cf push` would fail. This commit fixes that.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6009

fureigh committed:
Change remaining CloudFoundry URL to cloud.gov

...so the installation instructions are accurate.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6009

fureigh committed:
Update settings.php to work with aws-rds services

Previously the app couldn't connect to the database because the
settings.php file 1) wouldn't find aws-rds services and then 2)
looked for a couple of VCAP values that are no longer accurate.
This commit fixes those problems and enables the app to connect so
installation can proceed.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6009

fureigh commented:
Ready for review! My sample app is [up and running](https://cf-ex-drupal7.app.cloud.gov); please deploy your own as part of testing this. :)

To be handled in separate tickets:
* Make this (whether as https://github.com/18F/cf-ex-drupal/ or separately) discoverable from the general cloud.gov documentation.
* Diff this against standard D7 core to identify what, if anything, we'd need to document to be able to generalize this to be applicable to already-extant sites.

fureigh commented:
Changed PR title from `Update README to be cloud.gov-specific` to `Update this sample app to work on cloud.gov` to reflect that the scope extends beyond documentation.

jcscottiii commented on README.md:
````diff
@@ -1,43 +1,43 @@
+    ```bash
+    git clone https://github.com/18F/cf-ex-drupal.git cf-ex-drupal
+    cd cf-ex-drupal
+    ```
````
we need to add an instruction to change the hostname of the app in the manifest.yml. upon pushing at this point, i received the error:

```
Creating app cf-ex-drupal in org sandbox-gsa / space jamesc.scott as jamesc.scott@gsa.gov...
OK

Using route cf-ex-drupal.app.cloud.gov
Binding cf-ex-drupal.app.cloud.gov to cf-ex-drupal...
FAILED
The route cf-ex-drupal.app.cloud.gov is already in use.
TIP: Change the hostname with -n HOSTNAME or use --random-route to generate a new route and then push again.
```

jcscottiii commented on README.md:
````diff
@@ -1,43 +1,43 @@
 
-  ```bash
-  cf push
-  ```
````
we need to add an instruction to change the hostname of the app in the manifest.yml. upon pushing at this point, i received the error:

```
Creating app cf-ex-drupal in org sandbox-gsa / space jamesc.scott as jamesc.scott@gsa.gov...
OK

Using route cf-ex-drupal.app.cloud.gov
Binding cf-ex-drupal.app.cloud.gov to cf-ex-drupal...
FAILED
The route cf-ex-drupal.app.cloud.gov is already in use.
TIP: Change the hostname with -n HOSTNAME or use --random-route to generate a new route and then push again.
```

jcscottiii replied:
i pushed up the small modification to the manifest.yml (since i made it locally anyhow) but i didn't write instructions :/

fureigh replied:
Ah, of course. I'll add instructions etc. Thanks!

brittag replied:
Ha, the `--random-route` option is pretty fun if you haven't run into it before - when you set that flag, it automatically adds a couple random words to the end of the hostname, like `cf-ex-drupal-fancy-sloth`, so that you don't have to worry about it.

Commit:
Add host attribute to manifest file

jcscottiii commented on README.md:
```diff
@@ -1,43 +1,43 @@
-## CloudFoundry PHP Example Application: Drupal 
+## cloud.gov example application: Drupal 7
```
a question, should change the name of this repo to `cf-ex-drupal-7`?

jcscottiii replied:
i only say that if we think the setup for drupal 8 vs drupal 7 would be very different.

jcscottiii replied:
if we do choose to keep them in the same repo, we will probably have to create two top level sub-directories (e.g. `druapl7` and `drupal8`) so that you could `cd` into either then run `cf push`

jcscottiii replied:
also we can cross that bridge when we get there too. just thoughts

fureigh replied:
Good question... I'm inclined to leave it as-is and, if/as we expand it to include other versions, move their files to subfolders while keeping just one README.

fureigh replied:
@jcscottiii Looks like I wrote my reply to 'a question' while you were writing the other comments. :) +1 to creating two-level subdirectories once we have Drupal 8 ready to deploy.

fureigh replied:
@jcscottiii This raises the question, though: should we move this to be part of the `cf-hello-worlds` repo?

fureigh committed:
Add instructions for changing host attribute

This is intended to reduce the likelihood of false failures due to
duplicate hostnames.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6009

fureigh committed:
Copy edit: disambiguate 'it'

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6009

fureigh commented:
@jcscottiii Ready for another quick look.

fureigh commented:
Merged, but if you have further feedback, @jcscottiii, lmk.

jcscottiii commented:
LGTM

fureigh commented:
@brittag Thanks for the `--random-route` suggestion! I hadn't encountered that before but it sounds perfect for this. I'll update the README accordingly.
